### PR TITLE
MSI-1664: Decimal qty could be added to cart when it's not configured for product.

### PR DESCRIPTION
**Pull request by p-bystritsky**
### Description

'is_qty_decimal' Product value is now taken into consideration while adding decimal Product quantity to Cart.

### Fixed Issues (if relevant)

1. magento-engcom/msi#1664: Decimal qty could be added to cart when it's not configured for product.

### Manual testing scenarios

1. Simple product created with turned 'Off' - 'Use decimals' in Advanced inventory
2. Open product page on frontend
3. Set qty 1.34 and add it to shopping cart
4. To shopping cart should be added 1 item

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1665) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento-engcom/msi?pullRequest=1665) before we can accept your contribution.<br/>**3** out of **11** committers have signed the CLA.<br/><br/>:white_check_mark: naydav<br/>:white_check_mark: TomashKhamlai<br/>:white_check_mark: VitaliyBoyko<br/>:x: rogyar<br/>:x: chrom<br/>:x: pfantini<br/>:x: mage2pratik<br/>:x: TBlindaruk<br/>:x: yaroslav-zenin<br/>:x: magento-engcom-team<br/>:x: dmytro-ch

File: app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php [hunk 1]
```diff
@@ -111,6 +119,16 @@ public function aroundCheckQuoteItemQty(
         $result = $this->objectFactory->create();
         $result->setHasError(false);
 
+        $stockItem = $this->stockRegistryProvider->getStockItem($productId, $scopeId);
+        $result->setItemIsQtyDecimal($stockItem->getIsQtyDecimal());
+        if (!$stockItem->getIsQtyDecimal()) {
+            $result->setHasQtyOptionUpdate(true);
+            $itemQty = intval($itemQty);
```
Comment (phoenix128) [inline app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php]: Shouldn't be `(int) $itemQty` according to new guidelines?

**Review by phoenix128 — CHANGES_REQUESTED** [review]

**Comment by phoenix128** [comment]
FYI: I found the same problem yesterday night and it is also present in 2.3 for configurable products. It worth to investigate if the root cause in M2.3 instead of MSI.

**Comment by phoenix128** [comment]
FYI here is an alternate solution: https://github.com/magento-engcom/msi/pull/1671 .
Not sure what is the more appropriate for the MSI logic, but I think putting in the validation chain is a better way.

File: app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/InventorySales/Plugin/StockState/CheckQuoteItemQtyPlugin.php]: Fixed

**Comment by p-bystritsky** [comment]
> I think putting in the validation chain is a better way.

Yes, you are right.

**Pull request closed without merging**